packages/screens/LaunchpadERC20/LaunchpadERC20Sales/LaunchpadERC20CreateSaleForm.tsx

```diff
@@ -0,0 +1,190 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import React, { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { TouchableOpacity, View } from "react-native";
+
+import { LaunchpadERC20CreateSaleFooter } from "./LaunchpadERC20CreateSaleFooter";
+import { useCreateSaleState, zodCreateSaleForm } from "../hooks/useCreateSale";
+
+import { BrandText } from "@/components/BrandText";
+import ToggleButton from "@/components/buttons/ToggleButton";
+import { TextInputCustom } from "@/components/inputs/TextInputCustom";
+import { SpacerColumn } from "@/components/spacer";
+import useSelectedWallet from "@/hooks/useSelectedWallet";
+import { neutral77 } from "@/utils/style/colors";
+import { fontSemibold14, fontSemibold20 } from "@/utils/style/fonts";
+
+// TODO: ADD A WAY TO JUST DROP A CSV FILE OR ENTER A LIST OF ADDR AND IT WILL COMPUTE THE MERKLE ROOT FOR YOU
+// TODO: ADD A DATE PICKER FOR THE START TIMESTAMP AND END TIMESTAMP
+export const CreateSaleForm: React.FC = () => {
+  const {
+    actions: { goNextStep, setSale },
+    createSaleForm: formData,
+  } = useCreateSaleState();
+  const selectedWallet = useSelectedWallet();
+  const caller = selectedWallet?.address;
+  const { handleSubmit, formState, setValue, watch } = useForm({
+    resolver: zodResolver(zodCreateSaleForm),
+    defaultValues: formData,
+  });
+
+  const { errors } = formState;
+  useEffect(() => {
+    if (!caller) {
+      // TODO: would be better to not allow this corner case, aka do something smarter when no wallet is connected
+      return;
+    }
+    setValue("caller", caller);
+  }, [setValue, caller]);
+
+  if (!caller) {
+    return <BrandText>Connect a wallet to create a sale</BrandText>;
+  }
+
+  const values = watch();
+  return (
+    <View style={{ width: "100%", maxWidth: 480, margin: "auto" }}>
+      <BrandText style={fontSemibold20}>Sale Informations</BrandText>
+
+      <SpacerColumn size={1} />
+
+      <BrandText style={[fontSemibold14, { color: neutral77 }]}>
+        Information about your Sale
+      </BrandText>
+
+      <SpacerColumn size={2.5} />
+
+      <TextInputCustom
+        label="Token Name * (The token have to be owned by the caller)"
+        name="name"
+        fullWidth
+        placeholder="Type the token name here..."
+        variant="labelOutside"
+        onChangeText={(text) => setValue("tokenName", text)}
+        value={values.tokenName}
+        error={errors.tokenName?.message}
+      />
```

Comment:
```suggestion
      <TextInputCustom<TCreateSaleForm>
        label="Token Name (The token have to be owned by the caller)"
        name="tokenName"
        fullWidth
        placeholder="Type the token name here..."
        variant="labelOutside"
        control={control}
        rules={{required: true}}
      />
```
The asterix ![image](https://github.com/user-attachments/assets/a3cbdf23-bd86-4fdf-83c5-eff2506b7acc) is broken on `TextInputCustom.tsx`. We add `rules={{required: true}}` ONLY to display the asterix. It's decorative